python_modules/libraries/dagster-k8s/dagster_k8s/pipes.py

```diff
@@ -115,11 +124,14 @@ def __init__(
             )
             or PipesEnvContextInjector()
         )
-
         self.message_reader = (
             check.opt_inst_param(message_reader, "message_reader", PipesMessageReader)
             or PipesK8sPodLogsMessageReader()
         )
+        self.load_incluster_config = check.bool_param(
+            load_incluster_config, "load_incluster_config"
+        )
+        self.kubeconfig_file = check.opt_str_param(kubeconfig_file, "kubeconfig_file")
```

Comment:
* I briefly had changed the defaulting behavior for `load_incluster_config` to be dependent on `kubeconfig_file` but it didnt feel great when i went to document it so i walked it back. Maybe its fine. 

* its not clear to me that expecting the orchestration process to be in `kubernetes` is as safe an assumption as it is in our other APIs. Its still probably the majority case

Reply:
bonus of the current setup that you have is that its consistent with other k8s classes, but i'll throw in my vote for one kubeconfig_file parameter - seems like it results in fewer invalid/inconsistent states as per below comment